bookstore/handlers.py

```diff
@@ -44,39 +44,64 @@ def load_jupyter_server_extension(nb_app):
     web_app = nb_app.web_app
     host_pattern = '.*$'
 
-    # Always enable the version handler
-    base_bookstore_pattern = url_path_join(web_app.settings['base_url'], '/bookstore')
-    base_bookstore_api_pattern = url_path_join(web_app.settings['base_url'], '/api/bookstore')
-    web_app.add_handlers(host_pattern, [(base_bookstore_api_pattern, BookstoreVersionHandler)])
+    base_url = web_app.settings['base_url']
+
     bookstore_settings = BookstoreSettings(parent=nb_app)
-    web_app.settings['bookstore'] = {
-        "version": version,
-        "validation": validate_bookstore(bookstore_settings),
-    }
-
-    check_published = [
-        web_app.settings['bookstore']['validation'].get("bookstore_valid"),
-        web_app.settings['bookstore']['validation'].get("publish_valid"),
-    ]
-
-    if not all(check_published):
-        nb_app.log.info("[bookstore] Not enabling bookstore publishing, endpoint not configured")
+    validation = validate_bookstore(bookstore_settings)
+    web_app.settings['bookstore'] = {"version": version, "validation": validation}
+    handlers = build_handlers(nb_app.log, base_url, validation)
+    web_app.add_handlers(host_pattern, handlers)
+
+
+def build_handlers(log, base_url, validation):
+    """Utility for determinine which handlers should be added to the webapp.
```

Comment:
```suggestion
    """Utility for determine which handlers should be added to the webapp.
```

Reply:
I think it needs to be "for determining" or "to determine", but I'll just accept this as is and fix that in a follow-up commit.